pandas/_libs/tslibs/util.pxd

```diff
@@ -51,7 +52,7 @@ cdef inline int64_t get_nat() noexcept:
 # --------------------------------------------------------------------
 # Type Checking
 
-cdef inline bint is_integer_object(object obj) noexcept nogil:
```

Comment:
We previously declared is_timedelta64_object / is_datetime64_object to be nogil, which cascaded through the rest of the functions in this pxd file. However, the upstream function is not considered nogil, with some discussion coming back to here:

https://github.com/numpy/numpy/pull/16266#discussion_r426223194

I think its a mistake for _any_ of the functions in this module that touch CPython objects to be declared nogil, unless something more recent in CPython has declared these functions as such (haven't looked)

Reply:
can we just rip this out since we're using the cnp version everywhere?

Reply:
oh never mind, i see you did that where relevant below

Reply:
You are talking about the nogil right? Yea I did the minimum required for this PR. There are still other functions in util.pxd where it should probably be removed